server/planning/planning_locks.py

```diff
@@ -111,6 +121,54 @@ def _get_planning_module_locks():
     return locks
 
 
+def _prepare_query() -> SearchRequest:
+    """Prepare a SearchRequest object for querying locked items
+
+    Returns:
+        SearchRequest: A SearchRequest object configured with:
+            - Query filter for items with lock_session field
+            - Projection fields from PROJECTED_FIELDS
+            - Page 1 with 1000 results per page
+    """
+    elastic_query = ESQuery()
+    elastic_query.query.must.append(field_exists("lock_session"))
+
+    return SearchRequest(
+        args=SearchArgs(
+            source=elastic_query.generate_query_dict(),
+            projections=json.dumps(PROJECTED_FIELDS),
```

Comment:
You can provide the [projection argument](https://github.com/superdesk/superdesk-core/blob/async/superdesk/core/types/search.py#L163) directly to the SearchRequest top level variable (the args.projections is there for compatibility with the way Eve did things).

And there you can simply use a list of strings, rather than converting to a json string.

Reply:
You can specifically use the [elastic attribute](https://github.com/superdesk/superdesk-core/blob/async/superdesk/core/types/search.py#L169) of the SearchRequest directly as well, and the elastic layer will use that. This makes it a bit cleaner and easier to read, i.e.

```python
req = SearchRequest(
    projections=PROJECTED_FIELDS,
    page=1,
    max_results=1000,
    elastic=ESQuery(
        query=ESBoolQuery(
            must=[field_exists("lock_session")]
        )
    ),
)
```

We could probably clean it up a bit, so you don't have to use ESQuery nor ESBoolQuery, but for now the above should do the trick

Reply:
> the `args.projections` is there for compatibility with the way Eve did things

ahh good to know. I didn't know which one should be the preferred one to use. I will add a comment there because I think it is not really clear why we have both, or maybe I've missed that. Thanks!